### PR TITLE
feature: swipable pokemon items

### PR DESCRIPTION
**Pull request by tkumor3**
https://github.com/user-attachments/assets/76ac76a6-91fa-4576-a766-8436538c3f27

### CODE REVIEW

File: src/components/Swipeable/Like.tsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
+import React, { PropsWithChildren, useRef } from "react";
+import { Animated, Text, StyleSheet } from "react-native";
+import { RectButton } from "react-native-gesture-handler";
+import Swipeable, {
+  SwipeableProps,
+} from "react-native-gesture-handler/Swipeable";
+import { useLikeContext } from "../../contexts/LikedContext";
+import { useSwipeable } from ".";
+
+const renderActions: (
```
Comment (mironiasty) [inline src/components/Swipeable/Like.tsx:10]: Could this be defined as component? 🤔

File: src/components/Swipeable/index.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+type SwipeableContextType = {
+  onOpenNextSwipeable: (next: React.RefObject<SwipeableLib | null>) => void;
+};
+
+const SwipeableContext = createContext<SwipeableContextType>({
+  onOpenNextSwipeable: () => {},
+});
+
+export const Provider = ({ children }: PropsWithChildren) => {
```
Comment (mironiasty) [inline src/components/Swipeable/index.tsx:19]: I think you can achieve this logic without context - by having `ref` defined in PokemonList and passed it to each single `SwipeableLike`.

File: src/components/Swipeable/index.tsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  const openedSwipeableRef = useRef<SwipeableLib | null>(null);
+
+  const onOpenNextSwipeable = useCallback(
```
Comment (mironiasty) [inline src/components/Swipeable/index.tsx:22]: Should this method be called `closeOtherSwipeable`? Or `onOpenSwipeable`.

**Review by mironiasty — APPROVED** [review]

File: src/components/Swipeable/index.tsx (hunk 3 above)
Comment (tkumor3) [reply src/components/Swipeable/index.tsx]: IMO it should be called`onOpenSwipeable` 👍

**Pull request merged**